src/libraries/AccountAccessParser.sol

```diff
@@ -239,6 +241,127 @@ library AccountAccessParser {
+
+            // Process each diff and apply normalization logic.
+            for (uint256 j = 0; j < diffs.length; j++) {
+                StateDiff memory diff = diffs[j];
+                bool shouldInclude = true;
+
+                // 1. If the state change is an EOA nonce increment, remove it.
+                if (isEOANonceIncrement(account, diff)) {
+                    shouldInclude = false;
+                }
+                // 2. Remove Gnosis Safe nonce increment and approve hash changes.
+                else if (isGnosisSafe(account)) {
+                    // 2.1 Nonce increment.
+                    if (isGnosisSafeNonceIncrement(diff)) {
+                        shouldInclude = false;
+                    }
+                    // 2.2 Setting an approve hash in storage.
+                    else if (isGnosisSafeApproveHash(diff)) {
+                        shouldInclude = false;
+                    }
+                }
+                // 3. If the slot contains a timestamp, normalize it to zeroes.
+                else if (slotContainsTimestamp(account, diff)) {
+                    diff = normalizeTimestamp(diff);
+                }
+
+                // Include the diff in our normalized array if it should be included.
+                if (shouldInclude) {
+                    normalizedChanges[normalizedCount] = TempStateChange({
+                        who: account,
+                        slot: diff.slot,
+                        firstOld: diff.oldValue,
+                        lastNew: diff.newValue
+                    });
+                    normalizedCount++;
+                    require(normalizedCount < maxStateChanges, "AccountAccessParser: Max state changes reached");
+                }
+            }
+        }
+
+        // Create the final array with the correct size.
+        TempStateChange[] memory finalArray = new TempStateChange[](normalizedCount);
+        for (uint256 i = 0; i < normalizedCount; i++) {
+            finalArray[i] = normalizedChanges[i];
+        }
+
+        // Return keccak256 hash of the abi-encoded normalized array.
+        return keccak256(abi.encode(finalArray));
+    }
+
+    /// @notice Checks if the state diff represents an EOA nonce increment
+    function isEOANonceIncrement(address _account, StateDiff memory _diff) internal view returns (bool) {
+        uint256 codeSize = _account.code.length;
+        return codeSize == 0 && _diff.slot == bytes32(0) && uint256(_diff.newValue) == uint256(_diff.oldValue) + 1;
+    }
+
+    /// @notice Checks if the state diff represents a Gnosis Safe nonce increment
+    function isGnosisSafeNonceIncrement(StateDiff memory _diff) internal pure returns (bool) {
+        // In Gnosis Safe, the nonce is stored at slot 5. See `GnosisSafeStorage.sol` to verify.
+        return _diff.slot == GNOSIS_SAFE_NONCE_SLOT && uint256(_diff.newValue) == uint256(_diff.oldValue) + 1;
```

Comment:
I think this is a stronger check right? Someone might have a state override applied that's > 1 ahead of the current nonce (for whatever reason). 
```suggestion
        return _diff.slot == GNOSIS_SAFE_NONCE_SLOT && uint256(_diff.newValue) >= uint256(_diff.oldValue) + 1;
```

Reply:
I believe the state override would get applied at the start, before the increment, so we'd still see the nonce only increment by 1?